### DISCUSSION TITLE
Exact time for the player "you looked up this name recently"

### DISCUSSION THREAD

**Discussion by isXander** [HypixelDev/PublicAPI#408] [Q&A] [answered]
I am trying to make my program work for the new limit for the player endpoint. My program needs to be as up to date as possible and I would like to know exactly how long the cooldown lasts before being able to send the request again.

**Comment by shadowtime2000** [comment]
I think this would fit as a GH discussion in the Q & A category cc @ConnorLinfoot

**Comment by ConnorLinfoot** [comment, marked as answer]
Right now it's up to 5 minutes, but I don't want to guarantee that it won't change (although we don't plan on it currently). I would advise caching name to UUID lookups for at least an hour, or use another service such as Mojang API or [PlayerDB](https://playerdb.co/)

**Reply by robere2** [reply]
Just an idea, would it be possible to provide this data in the response? Something like:
```json
{"success":false,"cause":"You have already looked up this name recently","cooldown":300000}
```
or, if it's not easy to add extra properties, even something like:
```json
{"success":false,"cause":"You have already looked up this name recently. Cooldown: 300000ms"}
```

**Reply by AzureAaron** [reply]
> Just an idea, would it be possible to provide this data in the response? Something like:
> 
> ```json
> {"success":false,"cause":"You have already looked up this name recently","cooldown":300000}
> ```
> 
> or, if it's not easy to add extra properties, even something like:
> 
> ```json
> {"success":false,"cause":"You have already looked up this name recently. Cooldown: 300000ms"}
> ```

There could also be an array added of which usernames you’re on cooldown for.

**Reply by BansonVuong** [reply]
It's more efficient to convert IGN to UUID before passing it because there is no cooldown for UUID lookups, unless you're hitting the rate limit.

**Comment by db1218** [comment]
For those who did not understand this at first like me:

This new throttle only applies to when using the `name` key.
You are able to still use `uuid` key without this throttle.

You can then use one of the methods mentioned above:
> I would advise caching name to UUID lookups for at least an hour, or use another service such as Mojang API or [PlayerDB](https://playerdb.co/)

**Reply by isXander** [reply]
For anyone who comes across this and is looking for a good solution for caching, consider Google Guava Cache. You can set it up to clear after time.

**Comment by BansonVuong** [comment]
Don't use `name`. Use Mojang's API and convert the IGN into a UUID before passing it, UUID lookups aren't affected by the cooldown.
